Java/libraries/recognizers-text-date-time/src/main/java/com/microsoft/recognizers/text/datetime/parsers/BaseTimeParser.java

```diff
@@ -0,0 +1,352 @@
+            }
+
+            // get second
+            String secStr = match.getGroup(Constants.SecondGroupName).value.toLowerCase();
+            if (!StringUtility.isNullOrEmpty(secStr)) {
+                second = Integer.parseInt(secStr);
+                hasSec = true;
+            }
+        }
+
+        // Adjust by desc string
+        String descStr = match.getGroup(Constants.DescGroupName).value.toLowerCase();
+
+        // ampm is a special case in which at 6ampm = at 6
+        if (isAmDesc(descStr, match)) {
+            if (hour >= Constants.HalfDayHourCount) {
+                hour -= Constants.HalfDayHourCount;
+            }
+
+            if (!checkRegex(config.getUtilityConfiguration().getAmPmDescRegex(), descStr)) {
+                hasAm = true;
+            }
+
+        } else if (isPmDesc(descStr, match)) {
+            if (hour < Constants.HalfDayHourCount) {
+                hour += Constants.HalfDayHourCount;
+            }
+
+            hasPm = true;
+        }
+
+        // adjust min by prefix
+        String timePrefix = match.getGroup(Constants.PrefixGroupName).value.toLowerCase();
+        if (!StringUtility.isNullOrEmpty(timePrefix)) {
+            PrefixAdjustResult prefixResult = config.adjustByPrefix(timePrefix, hour, minute, hasMin);
+            hour = prefixResult.hour;
+            minute = prefixResult.minute;
+            hasMin = prefixResult.hasMin;
+        }
+
+        // adjust hour by suffix
+        String timeSuffix = match.getGroup(Constants.SuffixGroupName).value.toLowerCase();
+        if (!StringUtility.isNullOrEmpty(timeSuffix)) {
+            SuffixAdjustResult suffixResult = config.adjustBySuffix(timeSuffix, hour, minute, hasMin, hasAm, hasPm);
+            hour = suffixResult.hour;
+            minute = suffixResult.minute;
+            hasMin = suffixResult.hasMin;
+            hasAm = suffixResult.hasAm;
+            hasPm = suffixResult.hasPm;
+        }
+
+        if (hour == 24) {
+            hour = 0;
+        }
+
+        StringBuilder timex = new StringBuilder(String.format("T%02d", hour));
+        
+        if (hasMin) {
+            timex.append(String.format(":%02d", minute));
+        }
+        
+        if (hasSec) {
+            timex.append(String.format(":%02d", second));
+        }
+
+        result.setTimex(timex.toString());
+
+        if (hour <= Constants.HalfDayHourCount && !hasPm && !hasAm && !hasMid) {
+            result.setComment(Constants.Comment_AmPm);
+        }
+
+        LocalDateTime resultTime = DateUtil.safeCreateFromMinValue(year, month, day, hour, minute, second);
+        result.setFutureValue(resultTime);
+        result.setPastValue(resultTime);
+
+        result.setSuccess(true);
+
+        return result;
+    }
+
+    private boolean isAmDesc(String descStr, Match match) {
+        return checkRegex(config.getUtilityConfiguration().getAmDescRegex(), descStr)
+        || checkRegex(config.getUtilityConfiguration().getAmPmDescRegex(), descStr)
```

Comment:
Nit: Let's fix the linting rules to keep operators at the end of the previous line.

Reply:
@JuanAr this can come in one of the next PRs.